api/bootstrap.proto

```diff
@@ -19,6 +19,101 @@ import "google/protobuf/wrappers.proto";
 
 import "validate/validate.proto";
 
+// Bootstrap :ref:`configuration overview <config_overview_v2_bootstrap>`.
+message Bootstrap {
+  // Node identity to present to the management server and for instance
+  // identification purposes (e.g. in generated headers).
+  Node node = 1 [(validate.rules).message.required = true];
+
+  message StaticResources {
+    // Static :ref:`Listeners <envoy_api_msg_Listener>`. These listeners are
+    // available regardless of LDS configuration.
+    repeated Listener listeners = 1;
+
+    // If a network based configuration source is specified for :ref:`cds_config
+    // <envoy_api_field_Bootstrap.DynamicResources.cds_config>`, it's necessary
+    // to have some initial cluster definitions available to allow Envoy to know
+    // how to speak to the management server. These cluster definitions may not
+    // use :ref:`EDS <arch_overview_dynamic_config_sds>` (i.e. they should be static
+    // IP or DNS-based).
+    repeated Cluster clusters = 2;
+
+    // [#not-implemented-hide:]
+    repeated Secret secrets = 3;
+  }
+  // Statically specified resources.
+  StaticResources static_resources = 2;
```

Comment:
by the use after define notion, shouldn't this go above the definition of the message? I know the hardcore proto people would disagree.

Reply:
ignore please. I just checked the docs. It looks fine.